### PR TITLE
MCnet proposal tweaks + Sherpa proposal & Durham University organisation

### PR DESCRIPTION
**Pull request by agbuckley**
Cf. title!

### CODE REVIEW

**Comment by agheata** [comment]
@agbuckley please assign @agheata , @xvallspl and @aflp91 as reviewers when this is ready

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+project: MCnet
+year: 2020
+organization:
+  - UDUR
+---
+
+# Description
+
+SHERPA is a Monte Carlo event generator for collider-based particle physics experiments such as the ones perfomed at CERN's Large Hadron Collider (LHC).  It simulates full events, starting with the incident protons and finishing with the intricate patterns of the produced high-energetic particles that will eventually interact with the detector, thereby producing the indispensible link between theory and data, and allowing the interpretation of the latter in the light of the former.  To describe such events a variety of techniques and models must be invoked, many of which are based on the perturbative expansion of Quantum Field Theory.  This is true in particular for the starting point of the simulations where the so-called hard process defines the overall structure of the events.  To describe this central bit of the simulation, cross sections must be evaluated as integrals of the corresponding transition amplitudes over the multi-dimensional phase space of the outgoing particles.  Due to the high dimensionality of the phase space, this integration is achieved through Monte Carlo methods.
```
Comment (aflp91) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: "perfomed" => "performed"
"indispensible" => "indispensable"

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+In the ongoing quest for increased precision of our understanding and modelling of the physics involved in the proton collisions, ever higher orders of the perturbation expansion must be evaluated.  This results in ever more complicated expressions for the transition amplitudes and an ever increasing number of final state particles and therefore the dimensionality of the phase space integration.  It is therefore not surprising that this first step of event simulation is the most intensive source of CPU consumption and becoming the bottleneck for the full exploitation of future LHC results.
```
Comment (aflp91) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: "first step of event simulation" => "first step of the event simulation" or "first step of event simulations"

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+This project will be co-supervised by Frank Krauss, SHERPA's primary author, and Tobias Weinzierl, an expert in HPC computations and their performance optimization.  The prject will focus on first steps to increase the efficiency of the SHERPA event generator, and in particular of its description of the hard process.  In a first step, the successful students will anlyse the code and derive a detailed performance profile. Given the embarrassingly parallel nature of the underlying algorithmics, the studies will focus on single-node performance. In Durham, Intel's Amplifier VTune and the Allinea tools are available for this task, as well as a suite of open source helpers such as Likwid.
```
Comment (aflp91) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: "prject" => "project"
"anlyse" => "analyse"

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * Improvements to SHERPA's performance, especially for the integration of cross sections for processes with multi-particle final-states
+ * First draft of an Exascale Roadmap for SHERPA
+
+## Requirements
+
+ * C++
+ * Profiling
+ * Performance optimisation
+ * Multithreading
+ * git + gitlab or similar CI configuration
```
Comment (aflp91) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: "gitlab" => "GitLab"

**Comment by agbuckley** [comment]
Thanks for the fixes. Addressed and pushed

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+With a detailed performance profile at hand, the students will work out an exascale optimisation roadmap. We expect this roadmap to discuss mainly generic strategies mapped onto the particular application domain. This includes the flattening and serialisation of data structures to avoid scattered memory accesses, the reordering of data accesses, where the multi-shoot character of Monte Carlo algorithms usually provides ample space for relatively simple performance optimization.
+
+## Task ideas
+
+The following tasks are envisaged:
+
+ * Overall performance profile of the SHERPA code
```
Comment (agheata) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: @agbuckley these ideas are a bit immaterial with respect to code development, they basically suggest running some profiling and writing a report. The program actually requires coding, accessible via a public repo. Cold you rephrase to reflect this?

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+## Expected results
+
+ * Improvements to SHERPA's performance, especially for the integration of cross sections for processes with multi-particle final-states
+ * First draft of an Exascale Roadmap for SHERPA
+
+## Requirements
+
+ * C++
+ * Profiling
```
Comment (agheata) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: Profiling -> experience with profiling tools? C++: which level?

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * Performance optimisation
+ * Multithreading
```
Comment (agheata) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: You will find not many candidates with experience on performance optimisation, and it is hard to demonstrate as skill.

File: _gsocproposals/2020/proposal_MCnetSherpa.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+This project will be co-supervised by Frank Krauss, SHERPA's primary author, and Tobias Weinzierl, an expert in HPC computations and their performance optimization.  The project will focus on first steps to increase the efficiency of the SHERPA event generator, and in particular of its description of the hard process.  In a first step, the successful students will analyse the code and derive a detailed performance profile. Given the embarrassingly parallel nature of the underlying algorithmics, the studies will focus on single-node performance. In Durham, Intel's Amplifier VTune and the Allinea tools are available for this task, as well as a suite of open source helpers such as Likwid.
+
+With a detailed performance profile at hand, the students will work out an exascale optimisation roadmap. We expect this roadmap to discuss mainly generic strategies mapped onto the particular application domain. This includes the flattening and serialisation of data structures to avoid scattered memory accesses, the reordering of data accesses, where the multi-shoot character of Monte Carlo algorithms usually provides ample space for relatively simple performance optimization.
```
Comment (agheata) [inline _gsocproposals/2020/proposal_MCnetSherpa.md]: What means "working out an exascale roadmap"? It has to be something concrete, students are supposed to code, not make roadmaps. Can you rephrase in this direction?

File: _gsocproposals/2020/proposal_MCnetSherpa.md (hunk 5 above)
Comment (agbuckley) [reply _gsocproposals/2020/proposal_MCnetSherpa.md]: It's not my project, but given the timescale I will try

File: _gsocproposals/2020/proposal_MCnetSherpa.md (hunk 7 above)
Comment (agbuckley) [reply _gsocproposals/2020/proposal_MCnetSherpa.md]: True , but that is what's needed and it's a code task. If the candidates aren't available, that's life

File: _gsocproposals/2020/proposal_MCnetSherpa.md (hunk 8 above)
Comment (agbuckley) [reply _gsocproposals/2020/proposal_MCnetSherpa.md]: I have attempted, but this would be better as an update from the project supervisor.

**Comment by agbuckley** [comment]
I am out of time, so have pushed with small fixes to address the requests. Hopefully Frank Krauss can supply some updates asap, but for now it would be good to get this on the website... especially as today is the deadline, and we're getting clashes in the mentors.md file every time I rebase!

File: _gsocproposals/2020/proposal_MCnetSherpa.md (hunk 6 above)
Comment (agbuckley) [reply _gsocproposals/2020/proposal_MCnetSherpa.md]: Done

File: _gsocproposals/2020/proposal_MCnetSherpa.md (hunk 5 above)
Comment (agbuckley) [reply _gsocproposals/2020/proposal_MCnetSherpa.md]: Done

**Comment by agbuckley** [comment]
PS. The modifications I made were based on the similar, approved MC optimisation project https://hepsoftwarefoundation.org/gsoc/2020/proposal_Madgraph5-Accelerator.html
